test/topology_custom/test_topology_upgrade_stuck.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+#
+
+import asyncio
+import logging
+import pytest
+import time
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import inject_error
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.conftest import skip_mode
+from test.topology.util import reconnect_driver, enter_recovery_state, \
+        delete_raft_data_and_upgrade_state, log_run_time, wait_until_upgrade_finishes as wait_until_schema_upgrade_finishes, \
+        wait_until_topology_upgrade_finishes, delete_raft_topology_state, wait_for_cdc_generations_publishing, \
+        check_system_topology_and_cdc_generations_v3_consistency
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
```

Comment:
I think we should run it in `dev` only for not slowing down CI too much. Especially since it boots a cluster in gossip mode sequentially.

Reply:
Done, skipped `debug` mode for this test